db/write_batch.cc

```diff
@@ -1715,6 +1715,10 @@ class MemTableInserter : public WriteBatch::Handler {
 
     MemTable* mem = cf_mems_->GetMemTable();
     auto* moptions = mem->GetImmutableMemTableOptions();
+    if (moptions->merge_operator == nullptr) {
```

Comment:
I think this is going to have problems with PermitUncheckedError

Reply:
Yes, the proposed refactoring in the other PR to `SeekToColumnFamily()` could help here to make it clearer whether `ret_status` was checked.